File: src/main/java/me/nepnep/nepbot/message/command/ICommand.java
package me.nepnep.nepbot.message.command;

import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public interface ICommand {
    void execute(List<String> args, GuildMessageReceivedEvent event);

    String getInvoke();

    Permission requiredPermission();

    Category getCategory();

    String getDescription();
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/ICommand.kt
package me.nepnep.nepbot.message.command

import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent

interface ICommand {
    fun execute(args: List<String>, event: GuildMessageReceivedEvent)

    fun getInvoke(): String

    fun getRequiredPermission() = Permission.MESSAGE_WRITE

    fun getCategory(): Category

    fun getDescription(): String
}